Makefile

```diff
@@ -210,32 +176,14 @@ compdb:
 tidy:
 	$(RUN) tidy
 
-.PHONY: clear_xcode_cache
-clear_xcode_cache:
-ifeq ($(BUILD), osx)
```

Comment:
We should continue to clear the derived data. (It’s possible to clear out data derived from building core and the SDK without clearing out data derived from gyp project generation.)

However, I don’t think the derived data is being laid down in ~/Library/Developer/Xcode/DerivedData/{mapboxgl-app,osxapp}-* anyways. It seems to be going to ~/Library/Developer/Xcode/DerivedData/osx-* instead, at least when you build inside Xcode. Maybe it goes in ~/Library/Developer/Xcode/DerivedData/{mapboxgl-app,osxapp}-* if you run `make run-osx`?